### PR TITLE
fix PILOT_ENABLE_RDS_CACHE flag not working

### PR DESCRIPTION
**Pull request by yingzhuivy**
**Please provide a description of this PR:**

When building RDS, cache get is always conducted not respecting the
PILOT_ENABLE_RDS_CACHE flag. This PR fixes that.

Cache add already respects the flag:
https://github.com/istio/istio/blob/1.14.3/pilot/pkg/networking/core/v1alpha3/httproute.go#L212

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/httproute.go [hunk 1]
```diff
@@ -375,9 +377,11 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 	// Get list of virtual services bound to the mesh gateway
 	virtualHostWrappers := istio_route.BuildSidecarVirtualHostWrapper(routeCache, node, push, servicesByName, virtualServices, listenerPort)
 
-	resource, exist := xdsCache.Get(routeCache)
-	if exist && !features.EnableUnsafeAssertions {
-		return nil, resource, routeCache
+	if features.EnableRDSCaching {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/httproute.go:380]: for get we donot need this

**Comment by yingzhuivy** [comment]
yes, disabling it helps with propagation delay. I posted my testing here: https://github.com/istio/istio/issues/40744

File: pilot/pkg/networking/core/v1alpha3/httproute.go (hunk 1 above)
Comment (yingzhuivy) [reply pilot/pkg/networking/core/v1alpha3/httproute.go:380]: why is that? for cds we do respect it for get as well
https://github.com/istio/istio/blob/1.14.3/pilot/pkg/networking/core/v1alpha3/cluster_builder.go#L1271

**Review by hzxuzhonghu — APPROVED** [review]
improve on not construct

**Pull request merged**